jerry-core/parser/regexp/re-compiler.c

```diff
@@ -249,6 +249,12 @@ re_parse_alternative (re_compiler_ctx_t *re_ctx_p, /**< RegExp compiler context
   uint32_t idx;
   re_bytecode_ctx_t *bc_ctx_p = re_ctx_p->bytecode_ctx_p;
   ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+#ifdef REGEXP_RECURSION_LIMIT
+  if (--re_ctx_p->recursion_counter == 0)
+  {
+    return ecma_raise_range_error ("RegExp executor recursion limit is exceeded.");
+  }
+#endif /* REGEXP_RECURSION_LIMIT */
```

Comment:
IMO not only `REGEXP_INCREASE_RECURSION_COUNTER` should be defined as a macro but the other recursion limit-related snippets as well. E.g., this snippet could be named `REGEXP_DECREASE_RECURSION_COUNTER` (or `REGEXP_DECREASE_AND_TEST_RECURSION_COUNTER` even), while `re_ctx.recursion_counter = REGEXP_RECURSION_LIMIT;` could be `REGEXP_INIT_RECURSION_COUNTER`.

(Alternatively, word order might be changed to have the constant part as a prefix: `REGEXP_RECURSION_COUNTER_{INIT,INCREASE,DECREASE_AND_TEST}`.)

Reply:
Good Idea! I will update the PR with these changes.